src/pybind/mgr/cephadm/module.py

```diff
@@ -2456,6 +2464,57 @@ def get_osd_by_id(self, osd_id: int) -> Optional[Dict[str, Any]]:
 
         return osd[0]
 
+    def set_global_osd_flag(self, flag: str) -> None:
+        # attempt to set an osd flag. If we get a nonzero return code,
+        # raise an OrchestratorError
+        rc, out, err = self.mon_command({
+            'prefix': 'osd set',
+            'key': f'{flag}',
+        })
+        if rc:
+            self.log.warning(
+                f"Setting osd flag {flag} failed with return code {rc}: {err}")
+            raise OrchestratorError(
+                f"Setting osd flag {flag} failed with return code {rc}: {err}")
+        else:
+            self.log.info(
+                f"Set global osd flag {flag}")
+
+    def unset_global_osd_flag(self, flag: str) -> None:
+        # attempt to unset an osd flag. If we get a nonzero return code,
+        # raise an OrchestratorError
+        rc, out, err = self.mon_command({
+            'prefix': 'osd unset',
+            'key': f'{flag}',
+        })
+        if rc:
+            self.log.warning(
+                f"Unsetting osd flag {flag} failed with return code {rc}: {err}")
+            raise OrchestratorError(
+                f"Unsetting osd flag {flag} failed with return code {rc}: {err}")
+        else:
+            self.log.info(
+                f"Unset global osd flag {flag}")
+
+    def get_set_global_osd_flags(self) -> List[str]:
```

Comment:
I find the name of this function confusing, it sort of sounds like it can both get and set flags. I'd go for something more like `get_current_global_osd_flags`.